src/Panel/Plugins.php

```diff
@@ -75,9 +83,30 @@ public function read(string $type): string
 	{
 		$dist = [];
 
-		foreach ($this->files() as $file) {
+		$files = $this->files();
+
+		// Filter out all index.js files that shouldn't be loaded because an index.dev.mjs exists
+		if ($type === 'js') {
+			$files = A::filter(
+				$files,
+				function ($file) {
+					$is_js = Str::endsWith($file, 'index.js');
+					$has_dev_mjs = F::exists(preg_replace('/\.js$/', '.dev.mjs', $file));
+
+					return ($is_js && $has_dev_mjs) === false;
```

Comment:
```suggestion
					$isJs = Str::endsWith($file, 'index.js');
					$hasDevMjs = F::exists(preg_replace('/\.js$/', '.dev.mjs', $file));

					return $isJs === false || $hasDevMjs === false;
```

Do we even need the `jsJs` check here though? Down below in the loop we already check the extension. Unless I miss something here, we could replace this with a check just for `.dev.mjs`, which will allow a one-liner arrow function.

Reply:
You're right, I think it should work. Only ~time~ tests will tell.